src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.cs

```diff
@@ -492,37 +493,37 @@ public ManagedObjectWrapperHolder(ManagedObjectWrapper* wrapper, object wrappedO
             public bool IsActivated => _wrapper->Flags.HasFlag(CreateComInterfaceFlagsEx.IsComActivated);
 
             internal ManagedObjectWrapper* Wrapper => _wrapper;
+
+            internal ComWrappers ComWrappers { get; }
         }
 
         internal sealed unsafe class ManagedObjectWrapperReleaser
         {
-            private ManagedObjectWrapper* _wrapper;
+            private ManagedObjectWrapperHolder _holder;
 
-            public ManagedObjectWrapperReleaser(ManagedObjectWrapper* wrapper)
+            public ManagedObjectWrapperReleaser(ManagedObjectWrapperHolder holder)
             {
-                _wrapper = wrapper;
+                _holder = holder;
             }
 
             ~ManagedObjectWrapperReleaser()
             {
-                IntPtr refCountedHandle = _wrapper->HolderHandle;
-                if (refCountedHandle != IntPtr.Zero && GCHandle.InternalGet(refCountedHandle) != null)
+                if (_holder.ComWrappers._managedObjectWrapperTable.TryGetValue(_holder.WrappedObject, out _))
```

Comment:
This condition will always be true. As long as you have a strong reference to the key of a conditional weak table, the GC will not clear the entry.

Reply:
Yeah as I slept on it I realized that this won't work. I've got another idea I'll try out today.